### PR TITLE
source-postgres: Exclude child partitions from discovery

### PR DESCRIPTION
**Pull request by willdonnelly**
**Description:**

This partially addresses https://github.com/estuary/connectors/issues/751 (specifically point #2) but the other bits of polish (documentation updates and better error messages) are still needed at some point.

This commit factors out table-enumeration from column-listing for simplicity (because the query was starting to get hard to follow) and to match what we did for MySQL and SQL Server just yesterday.

The new table-enumeration query is completely different from the old one but I believe it returns the same set of tables as before, minus child partitions. I ran a couple of the discovery tests to spot-check the results and added a couple new ones, so I think it is correct.

As with the other connectors, the column-listing query no longer tries to also filter for table eligibility. Its only job is to list every column of every table in the database.

**Workflow steps:**

Child partitions of a partitioned table should no longer be returned by discovery, which means that blindly refreshing the bindings of a capture should be safe in more circumstances.

- - -
This change is [Reviewable](https://reviewable.io/reviews/estuary/connectors/917)

### CODE REVIEW

File: source-postgres/discovery.go [hunk 1]
```diff
@@ -49,15 +53,15 @@ func (db *postgresDatabase) DiscoverTables(ctx context.Context) (map[string]*sql
...
 	for _, column := range columns {
 		var streamID = sqlcapture.JoinStreamID(column.TableSchema, column.TableName)
 		var info, ok = tableMap[streamID]
 		if !ok {
-			info = &sqlcapture.DiscoveryInfo{
-				Schema:    column.TableSchema,
-				Name:      column.TableName,
-				BaseTable: true, // PostgreSQL discovery queries only ever list 'BASE TABLE' entities
-			}
+			continue
```
Comment (williamhbaker) [inline source-postgres/discovery.go:64]: I think this should only happen if its a column in a system table, if I'm reading things right. If not, it might be worth logging something here since (I assume) it should never happen otherwise.

**Review by williamhbaker — APPROVED** [review]
LGTM

File: source-postgres/discovery.go (hunk 1 above)
Comment (willdonnelly) [reply source-postgres/discovery.go:64]: A system table or a child partition, now. This is basically the same logic we already used for primary-key aggregation: if the table isn't already in the list then an earlier discovery step has decided we don't care about it and should ignore this information.

Logging whenever this happens would be unusably spammy even for debug level IMO.
